data/data/bootstrap/files/usr/local/bin/installer-gather.sh

```diff
@@ -9,54 +9,108 @@ fi
 ARTIFACTS="/tmp/artifacts-${GATHER_ID}"
 mkdir -p "${ARTIFACTS}"
 
-exec &> >(tee "${ARTIFACTS}/gather.log")
+# The existence of the file located in BOOTSTRAP_IN_PLACE_BOOTSTRAP_PHASE_ARCHIVE_PATH is used
+# as indication that we're running inside a single-node bootstrap-in-place deployment post-pivot
+# master node, rather than a typical bootstrap machine. In the former case, we extract some of
+# the logs from said archive rather than actually collect them.
+BOOTSTRAP_IN_PLACE_BOOTSTRAP_PHASE_LOG_BUNDLE_NAME="log-bundle-bootstrap-in-place-pre-reboot"
+BOOTSTRAP_IN_PLACE_BOOTSTRAP_PHASE_ARCHIVE_PATH="/var/log/$BOOTSTRAP_IN_PLACE_BOOTSTRAP_PHASE_LOG_BUNDLE_NAME.tar.gz"
+if [[ -f ${BOOTSTRAP_IN_PLACE_BOOTSTRAP_PHASE_ARCHIVE_PATH} ]]; then
+    # single-node bootstrap-in-place deployment post-pivot master node log gathering
+
+    exec &> >(tee "${ARTIFACTS}/bootstrap-in-place-post-pivot-gather.log")
+
+    # Instead of gathering bootstrap logs, copy from the pre-preboot gather archive
+    echo "Found log bundle from bootstrap phase, running in bootstrap-in-place post-pivot mode"
+    tar -xzf ${BOOTSTRAP_IN_PLACE_BOOTSTRAP_PHASE_ARCHIVE_PATH}
+    cp -r ${BOOTSTRAP_IN_PLACE_BOOTSTRAP_PHASE_LOG_BUNDLE_NAME}/* "${ARTIFACTS}/"
```

Comment:
I like this direction and I think we can take it one step further.
I think that all things from the bootstrap log bundle should get a new home - ${ARTIFACTS}/${LOG_BUNDLE_NAME}.

I think this is the way to go for few reasons:

1. The log bundle collected while running from liveCD are defacto logs from a different node.
2. We won't lose any information (such as the crio service logs pre/post reboot).
3. It will simplify the code since there is no need for all the `rm` and `cp`.
4. This feature can work for any existing log bundle and not just for single node bootstrap-in-place.

While we want to have a similar log bundle to the regular installation flow we can't ignore that the bootstrap logic ran on "another node" earlier in the process, so instead of "hiding it" we should explicitly reflect that by placing it under a new home.

Reply:
I agree, https://github.com/eranco74/installer/pull/13 looks very promising. Can we get that pulled into https://github.com/openshift/installer/pull/4482? I have a few more thoughts that I'll add there.

Reply:
It'll be like 10 hours until @eranco74 will merge it so if you have comments specific to #13 please write them in #13 now so we can deal with them in our morning in 10h, because our timezones are out of sync

Reply:
Not specific to #13... but #13 might need some modificaitons. https://github.com/openshift/installer/pull/4482/files#r568957299